test/Microsoft.ML.Tests/ScenariosWithDirectInstantiation/TensorflowTests.cs

```diff
@@ -152,11 +262,52 @@ public void TensorFlowTransformMNISTConvTest()
             }
         }
 
+        [Fact]
+        public void TensorFlowTransformMNISTConvPipelineTest()
+        {
+            var model_location = "mnist_model/frozen_saved_model.pb";
+            var dataPath = GetDataPath("Train-Tiny-28x28.txt");
+
+            var pipeline = new LearningPipeline(seed: 1);
+            pipeline.Add(new Microsoft.ML.Data.TextLoader(dataPath).CreateFrom<MNISTData>(useHeader: false));
+            pipeline.Add(new Transforms.ColumnCopier() { Column = new[] { new CopyColumnsTransformColumn() { Name = "reshape_input", Source = "Placeholder" } } });
+            pipeline.Add(new TensorFlowScorer()
+            {
+                ModelFile = model_location,
+                OutputColumns = new[] { "Softmax", "dense/Relu" },
+                InputColumns = new[] { "Placeholder", "reshape_input" }
```

Comment:
It is required, this column is created in line 273 by the CopyColumns transform.
This input is required for computing the "dense/Relu" output. Actually, I am not sure why it is needed, since if I understand correctly, reshape_input is computed by the input layer Placeholder by simply reshaping from 28x28 to 784. @zeahmed , do you know why "Placeholder" is not enough to compute "dense/Relu"?

Reply:
This is the problem with the model. [https://github.com/tensorflow/models/blob/master/official/mnist/mnist.py](https://github.com/tensorflow/models/blob/master/official/mnist/mnist.py)

If you want to access the `features` (named `dense/Relu`) only then `reshape_input` is required if you want to access the `Softmax` then `Placeholder` is required. This is the problem with model. If you closely look at the model graph attached. You would observe two graph that are working in parallel. Having said that I think its a good model to test two inputs and two outputs.

If you think if its going to make an issue I can update the model.

---
In reply to: [216485826](https://github.com/dotnet/machinelearning/pull/857#discussion_r216485826) [](ancestors = 216485826)

Reply:
Also, i think we should rename the nodes properly to avoid strange names like `dense/Relu` etc....:)

---
In reply to: [216502475](https://github.com/dotnet/machinelearning/pull/857#discussion_r216502475) [](ancestors = 216502475,216485826)